### PR TITLE
SNOW-1063738 [Local Testing] Triage test failures in more integration test files 

### PR DESCRIPTION
**Pull request by sfc-gh-stan**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-1063738 

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   This PR triages all Local Testing test failures in 
  - test_dataframe_copy_into.py
  - test_dataframe_join_suite.py 
  - test_stored_procedure.py
  - test_udf.py
  
by marking tests that should apply to Local Testing but fail due to bug/unsupported feature to be skipped and tests that should not apply (i.e. SQL feature tests) to `xfail`.

### CODE REVIEW

File: tests/integ/scala/test_dataframe_join_suite.py
Comment (sfc-gh-aling) [inline tests/integ/scala/test_dataframe_join_suite.py:166]: jira for error experience: https://snowflakecomputing.atlassian.net/browse/SNOW-1235716

File: tests/integ/scala/test_dataframe_join_suite.py
Comment (sfc-gh-aling) [inline tests/integ/scala/test_dataframe_join_suite.py:770]: did you do a quick triage of the failure (error information) or you will put some context into the doc?

File: tests/integ/test_session.py
Comment (sfc-gh-aling) [inline tests/integ/test_session.py:309]: hopefully we don't need this if-else after completely mocking connector connection, I have added this test into JIRA SNOW-1368486

File: tests/integ/test_udf.py
Comment (sfc-gh-aling) [inline tests/integ/test_udf.py:183]: can this be written into `df.select(call_udf(...)`?

File: tests/integ/test_udf.py
Comment (sfc-gh-aling) [inline tests/integ/test_udf.py:437]: this one shall be skipif? I see the below vector udf tests are marked skipif

File: src/snowflake/snowpark/session.py
Comment (sfc-gh-stan) [reply src/snowflake/snowpark/session.py:380]: This is because we shouldn't store the password, see the live connection equivalent at: https://github.com/snowflakedb/snowpark-python/blob/7a42f03d6d3a60901b30453cfd01a3768fbdc483/src/snowflake/snowpark/_internal/server_connection.py#L159

File: tests/integ/conftest.py
Comment (sfc-gh-stan) [reply tests/integ/conftest.py:146]: `local_testing_mode` is only True in Local Testing jobs, when it's False the created session will be a live connection session.

File: tests/integ/test_stored_procedure.py
Comment (sfc-gh-stan) [reply tests/integ/test_stored_procedure.py:78]: We don't have plans to work on packaging so yea I can convert this to an xfail

File: tests/integ/test_stored_procedure.py
Comment (sfc-gh-stan) [reply tests/integ/test_stored_procedure.py:1371]: This is still blocked because if_not_exists is not supported, I'll update the reason.

File: tests/integ/test_udf.py
Comment (sfc-gh-stan) [reply tests/integ/test_udf.py:183]: There is a `call_udf` below, I think this test case is for SQL specifically.

File: src/snowflake/snowpark/session.py
Comment (sfc-gh-stan) [reply src/snowflake/snowpark/session.py]: This is a change that we need to fix RuntimeConfig for Local Testing, I removed this change and created SNOW-1370578 instead.

File: tests/integ/test_udf.py
Comment (sfc-gh-jrose) [inline tests/integ/test_udf.py:396]:
```suggestion
    # Query tags not supported in local testing.
    if not local_testing_mode:
```

**Review by sfc-gh-jrose — COMMENTED** [review]
I'd like to normalize any TODO strings to be in this format:
```
SNOW-XXXXXXX: description
```
Skip strings are probably safe, but I've seen linters generate warnings for TODO in comments and I think we should treat the strings like comments.

File: tests/integ/conftest.py
Comment (sfc-gh-aling) [reply tests/integ/conftest.py:146]: oh yes that's true, thanks for reminding me of that

**Review by sfc-gh-aling — APPROVED** [review]
LGTM, should we merge this PR after all other enabling test PRs get merged first as it's setting `CONNECTION_PARAMETERS["local_testing"] = local_testing_mode`

**Review by sfc-gh-jrose — APPROVED** [review]

**Pull request merged**
